### PR TITLE
chore(tests): Add npm test:unit task, runs unit and component tests only (no e2e)

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Introduced npm test task that will run unit and component tests only
### What issues does this PR fix or reference?
#2707 

### How to test this PR?
`yarn test:unit`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "compile:pull-request": "cross-env MODE=production npm run build && electron-builder build --publish never --config .electron-builder.config.cjs",
     "compile:current": "cross-env MODE=production npm run build && electron-builder build --config .electron-builder.config.cjs",
     "test": "npm run test:main && npm run test:preload && npm run test:preload-docker-extension && npm run test:renderer && npm run test:tools && npm run test:e2e && npm run test:extensions",
+    "test:unit": "npm run test:main && npm run test:preload && npm run test:preload-docker-extension && npm run test:renderer && npm run test:tools && npm run test:extensions",
```
Comment (benoitf) [inline package.json:43]: probably then "test" should be just

`"test" : "npm run test:unit && npm run test:e2e"`

**Review by benoitf — APPROVED** [review]

**Comment by odockal** [comment]
@benoitf PR check failing. After e2e task removal, there is missing build step that is used in extensions tests. I have added building into pr check (unit tests).

**Comment by benoitf** [comment]
I think this is an issue, we should not have to do yarn build

**Comment by benoitf** [comment]
probably related to https://github.com/containers/podman-desktop/issues/2451

**Comment by odockal** [comment]
> I think this is an issue, we should not have to do yarn build

We did before. It was part of e2e. Extensions tests were run after that, I guess, on purpose. But issue is already there: https://github.com/containers/podman-desktop/issues/2111

**Comment by odockal** [comment]
CI run failure:
```

 ✓ src/kind-installer.spec.ts  (3 tests | 1 skipped) 276ms
 ❯ src/create-cluster.spec.ts  (0 test)
 ❯ src/extension.spec.ts  (0 test)

⎯⎯⎯⎯⎯⎯ Failed Suites 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/create-cluster.spec.ts [ src/create-cluster.spec.ts ]
 FAIL  src/extension.spec.ts [ src/extension.spec.ts ]
Error: ENOENT: no such file or directory, open '/src-generated/contour.yaml'
```

**Comment by odockal** [comment]
@benoitf i have requested your review again since I have added a change (separate build step). fyi.

**Comment by benoitf** [comment]
@odockal I've created https://github.com/containers/podman-desktop/pull/2757 to avoid to do the build step

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -200,6 +200,9 @@ jobs:
       - name: Run formatter
         run: yarn format:check
 
+      - name: Run build
+        run: yarn build
+
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: @odockal  with https://github.com/containers/podman-desktop/pull/2757 merged you should be able to remove this

Comment (benoitf) [inline .github/workflows/pr-check.yaml]:
```suggestion
```

**Pull request merged**